File: 1_intro/2_edge_of_the_ocean/8_matrixElementsSum/matrixElementSum.js
function matrixElementsSum(matrix) {
    for (var r = 0, j = 0; j < matrix[0].length; j++) {
        for (var i = 0; i < matrix.length; i++) {
            if (matrix[i][j] === 0) break
            else r += matrix[i][j]
        }
    }
    return r;
}

File: 1_intro/2_edge_of_the_ocean/8_matrixElementsSum/matrixElementSum.lua
function matrixElementsSum(matrix)
  r = 0
  for j = 1, table.getn(matrix[1]) do
    for i = 1, table.getn(matrix) do
      if matrix[i][j] == 0 then break
      else
        r = r + matrix[i][j]
      end
    end 
  end
  return r
end